enterprise_access/apps/workflow/docs/overview.rst

```diff
@@ -0,0 +1,271 @@
+  For our workflow app, these classes represent the specific input or output data structures
+  for units of work. ``cattrs`` is used for structuring (converting dictionaries, often from JSON,
+  into ``attrs`` instances) and unstructuring (converting ``attrs`` instances back into dictionaries).
+
+``BaseInputOutput``
+  This class serves as a base for all specific input and output data classes.
+  It provides common functionality, such as ``from_dict()`` and ``to_dict()`` methods,
+  leveraging ``cattrs`` for conversion.
+
+Static ``KEY`` Attribute
+  Concrete input/output classes (defined by specific workflow implementations)
+  should have a static ``KEY`` string attribute. This key is used by the ``AbstractWorkflow`` model
+  to dynamically construct its composite input/output classes
+  and to correctly map data between the overall workflow and its individual steps.
+
+Validation
+  ``attrs`` validators are used to ensure data integrity
+  for the fields within these input/output classes.
+
+Abstract Unit of Work (``apps/workflow/models.py:AbstractUnitOfWork``)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is the most fundamental building block of the workflow pattern, representing a persistent set of
+work that can be executed. Both individual workflow steps and entire workflows are types of "units of work."
+
+Key Django Model Attributes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``uuid``
+  Primary key, a unique identifier for this unit of work instance.
+``input_data`` (``JSONField``)
+  Stores the input required for this unit, serialized as JSON.
+``output_data`` (``JSONField``)
+  Stores the result after execution, serialized as JSON.
+``created``, ``modified`` (from ``TimeStampedModel``)
+  Timestamps for creation and last modification.
+``is_removed`` (from ``SoftDeletableModel``)
+  For soft deletion.
+``succeeded_at``, ``failed_at``
+  Timestamps indicating successful or failed execution.
+``exception_message``
+  Stores the error message if execution failed.
+
+Key Class Attributes (to be defined by subclasses)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``input_class``
+  Specifies the ``attrs``-based class (inheriting from ``BaseInputOutput``)
+  used to structure and validate the ``input_data``. Defaults to an empty class.
+``output_class``
+  Specifies the ``attrs``-based class for the ``output_data``.
+  Defaults to an empty class.
+``exception_class``
+  The specific exception type to be raised by the ``execute`` method upon failure.
+  Defaults to ``UnitOfWorkException``.
+
+Key Properties & Methods
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+``input_object`` (property)
+  Deserializes ``self.input_data`` (JSON) into an instance of ``self.input_class``.
+``output_object`` (property)
+  Deserializes ``self.output_data`` (JSON) into an instance of ``self.output_class``.
+``process_input(self, accumulated_output=None, **kwargs)``
```

Comment:
Nit: this method and the three *_class attributes all share the need to be defined in a concrete subclass and should probably live together in the same definition list.

Reply:
In fact, input_object, output_object, and execute() might be extraneous information for this ADR which should IMO be more focused on interfaces, not internals.